<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="2" column="0">
     <layout class="QGridLayout" name="gridLayout_3">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="spacing">
       <number>0</number>
      </property>
      <item row="1" column="0">
       <widget class="QScrollBar" name="horizontalScrollBar">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="OpenGLRender2D" name="openGLWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QScrollBar" name="verticalScrollBar">
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>400</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="separator"/>
    <addaction name="action_file_load"/>
    <addaction name="action_file_save"/>
    <addaction name="separator"/>
    <addaction name="action_file_exit"/>
   </widget>
   <widget class="QMenu" name="menuSimulation">
    <property name="title">
     <string>Simulation</string>
    </property>
    <addaction name="action_simulation_single_step"/>
    <addaction name="action_simulation_validate"/>
    <addaction name="action_simulation_benchmark"/>
    <addaction name="separator"/>
    <addaction name="action_simulation_continuous"/>
   </widget>
   <widget class="QMenu" name="menuPresets">
    <property name="title">
     <string>Presets</string>
    </property>
    <addaction name="action_presets_glider"/>
    <addaction name="separator"/>
    <addaction name="action_presets_1k"/>
    <addaction name="action_presets_2k"/>
    <addaction name="separator"/>
    <addaction name="action_presets_4k"/>
    <addaction name="action_presets_8k"/>
    <addaction name="separator"/>
    <addaction name="action_presets_16k"/>
    <addaction name="separator"/>
    <addaction name="action_presets_32k"/>
    <addaction name="separator"/>
    <addaction name="action_presets_canvas"/>
   </widget>
   <widget class="QMenu" name="menuRenderer">
    <property name="title">
     <string>Renderer</string>
    </property>
    <addaction name="action_renderer_software"/>
    <addaction name="separator"/>
    <addaction name="action_renderer_GL"/>
    <addaction name="action_renderer_combined"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="action_view_fit"/>
    <addaction name="action_view_fancy"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuRenderer"/>
   <addaction name="menuSimulation"/>
   <addaction name="menuView"/>
   <addaction name="menuPresets"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_simulation_single_step">
   <property name="text">
    <string>Single Step</string>
   </property>
   <property name="shortcut">
    <string>F6</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New...</string>
   </property>
  </action>
  <action name="action_file_load">
   <property name="text">
    <string>Load...</string>
   </property>
  </action>
  <action name="action_file_save">
   <property name="text">
    <string>Save...</string>
   </property>
  </action>
  <action name="action_file_exit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="action_simulation_continuous">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Continuous Mode</string>
   </property>
   <property name="shortcut">
    <string>F4</string>
   </property>
  </action>
  <action name="action_simulation_validate">
   <property name="text">
    <string>Single Step Verify</string>
   </property>
   <property name="shortcut">
    <string>F7</string>
   </property>
  </action>
  <action name="action_renderer_software">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>CPU Texture Copy</string>
   </property>
  </action>
  <action name="action_renderer_GL">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>CUDA Buffer Copy</string>
   </property>
  </action>
  <action name="action_renderer_combined">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>CUDA Combined</string>
   </property>
  </action>
  <action name="action_view_fit">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Fit</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="action_view_fancy">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Fancy</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="action_simulation_benchmark">
   <property name="text">
    <string>Benchmark Kernel</string>
   </property>
   <property name="shortcut">
    <string>F8</string>
   </property>
  </action>
  <action name="action_presets_1k">
   <property name="text">
    <string>1K (1 MB)</string>
   </property>
  </action>
  <action name="action_presets_2k">
   <property name="text">
    <string>2K (4 MB)</string>
   </property>
  </action>
  <action name="action_presets_4k">
   <property name="text">
    <string>4K (16 MB)</string>
   </property>
  </action>
  <action name="action_presets_8k">
   <property name="text">
    <string>8K (64 MB)</string>
   </property>
  </action>
  <action name="action_presets_16k">
   <property name="text">
    <string>16K (256 MB)</string>
   </property>
  </action>
  <action name="action_presets_32k">
   <property name="text">
    <string>32K (1 GB)</string>
   </property>
  </action>
  <action name="action_presets_canvas">
   <property name="text">
    <string>Canvas Size</string>
   </property>
  </action>
  <action name="action_presets_glider">
   <property name="text">
    <string>GLIDER</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>OpenGLRender2D</class>
   <extends>QOpenGLWidget</extends>
   <header>openglrender2d.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
